pkg/repository/git_test.go

```diff
@@ -151,28 +203,49 @@ func TestGetSHA(t *testing.T) {
 		{
 			name: "get sha without any problem",
 			config: config{
-				repositoryGetter: fakeRepositoryGetter{
-					repository: &fakeRepository{
-						worktree: &fakeWorktree{
-							status: &fakeStatus{
-								isClean: true,
+				repositoryGetter: &fakeRepositoryGetter{
+					repository: []*fakeRepository{cleanRepo, cleanRepo},
+				},
+			},
+			expected: expected{
+				sha: plumbing.NewHash("test").String(),
+				err: nil,
+			},
+		},
+		{
+			name: "get empty sha when not clean",
+			config: config{
+				repositoryGetter: &fakeRepositoryGetter{
+					repository: []*fakeRepository{
+						{
+							worktree: &fakeWorktree{
+								status: oktetoGitStatus{
+									status: git.Status{
+										"test-file.go": &git.FileStatus{
+											Staging:  git.Modified,
+											Worktree: git.Unmodified,
+										},
+									},
+								},
 							},
+							head: plumbing.NewHashReference("test", plumbing.NewHash("test")),
 						},
-						head: plumbing.NewHashReference("test", plumbing.NewHash("test")),
 					},
 				},
 			},
 			expected: expected{
-				sha: plumbing.NewHash("test").String(),
+				sha: "",
 				err: nil,
 			},
 		},
 		{
 			name: "error getting repository",
 			config: config{
-				repositoryGetter: fakeRepositoryGetter{
-					repository: nil,
-					err:        assert.AnError,
+				repositoryGetter: &fakeRepositoryGetter{
```

Comment:
I had to refactor some stuff because this test and the one below were not working properly. They would pass but not for the reason we thought they would.

They were actualky the same code (expecting to test different things) which is a bit odd! 

They were both failing early in the function so not reaching their respective logic.

It should be fixed now!

Reply:
Thanks!!